src/neptune/metadata_containers/run.py

```diff
@@ -472,6 +475,11 @@ def _write_initial_attributes(self):
 
             dependency_strategy.log_dependencies(run=self)
 
+        track_uncommitted_changes(
+            git_ref=self._git_ref or GitRef(),
```

Comment:
What if someone doesn't want to track this? Why are we not taking into account the `GitRef.Disabled`?

Reply:
Check out `DiffTracker.from_git_ref` in `src/neptune/internal/utils/git.py`